psydac/core/bsplines.py

```diff
@@ -309,6 +315,8 @@ def collocation_matrix(knots, degree, periodic, normalization, xgrid, out=None):
     The collocation matrix :math:`C_ij = B_j(x_i)`, contains the
     values of each B-spline basis function :math:`B_j` at all locations :math:`x_i`.
     """
+    knots = np.asarray(knots)
+    xgrid = np.asarray(xgrid)
```

Comment:
As above

Reply:
Done in https://github.com/pyccel/psydac/commit/54fa1fb4c2afba79c77392308a193bad59161fee